rclpy/rclpy/node.py

```diff
@@ -100,6 +100,18 @@ def __init__(
 
         self.__executor_weakref = None
 
+        node_parameters = _rclpy.rclpy_get_node_parameters(Parameter, self.handle)
+        # Combine parameters from params files with those from the node constructor and
+        # use the set_parameters API so parameter events are published.
+        if node_parameters and initial_parameters:
+            params_by_name = {p.name: p for p in node_parameters}
+            params_by_name.update({p.name: p for p in initial_parameters})
+            self.set_parameters(params_by_name.values())
+        elif node_parameters:
+            self.set_parameters(node_parameters)
+        elif initial_parameters:
+            self.set_parameters(initial_parameters)
```

Comment:
At first glance, it looks like this could be simplified quite a bit by setting default values for node_parameters and initial_parameters.

(I didnt check the implementation so this is hypothetical)
But if:
`initial_parameters` defaulted to an empty list and `node_parameters` to an empty dict.
Could we get rid of the conditional logic and just have the following?
```
params_by_name = {p.name: p for p in node_parameters}
params_by_name.update({p.name: p for p in initial_parameters})
self.set_parameters(params_by_name.values())
```
I wouldnt expect a big performance hit if these are empty.